src/main/java/org/opensearch/flowframework/model/PipelineProcessor.java

```diff
@@ -0,0 +1,101 @@
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * This represents a processor associated with search and ingest pipelines in the {@link Template}.
+ */
+public class PipelineProcessor implements ToXContentObject {
+
+    /** The type field name for pipeline processors */
+    public static final String TYPE_FIELD = "type";
+    /** The params field name for pipeline processors */
+    public static final String PARAMS_FIELD = "params";
+
+    private final String type;
+    private final Map<String, String> params;
+
+    /**
+     * Create this processor with a type and map of parameters
+     * @param type the processor type
+     * @param params a map of params
+     */
+    public PipelineProcessor(String type, Map<String, String> params) {
+        this.type = type;
+        this.params = params;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject();
+        xContentBuilder.field(TYPE_FIELD, this.type);
+        xContentBuilder.field(PARAMS_FIELD);
+        Template.buildStringToStringMap(xContentBuilder, this.params);
+        return xContentBuilder.endObject();
+    }
+
+    /**
+     * Parse raw json content into a processor instance.
+     *
+     * @param parser json based content parser
+     * @return the parsed PipelineProcessor instance
+     * @throws IOException if content can't be parsed correctly
+     */
+    public static PipelineProcessor parse(XContentParser parser) throws IOException {
+        String type = null;
+        Map<String, String> params = new HashMap<>();
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+            switch (fieldName) {
+                case TYPE_FIELD:
+                    type = parser.text();
+                    break;
+                case PARAMS_FIELD:
+                    params = Template.parseStringToStringMap(parser);
+                    break;
+                default:
+                    throw new IOException("Unable to parse field [" + fieldName + "] in a pipeline processor object.");
+            }
+        }
+        if (type == null) {
+            throw new IOException("A processor object requires a type field.");
+        }
+
+        return new PipelineProcessor(type, params);
+    }
+
+    /**
+     * Get the processor type
+     * @return the type
+     */
+    public String type() {
+        return type;
+    }
+
+    /**
+     * Get the processor params
+     * @return the params
+     */
+    public Map<String, String> params() {
```

Comment:
Shouldn't we keep the type of params as `<String, Object>`?

Reply:
Pipeline processor params are just the user provided inputs, so a `Map<String,String>` makes sense. The directional idea here is to take these params and build the configuration, mapping the provided fields and values to some template. We'll probably have to create a set of classes that implement `ToXContentObject` for each type of pipeline processor and pass these params there.

Reply:
The problem with an object value is you need to define what it is.  We could allow nested maps but at some point you have to stop the recursion.